woocommerce/woocommerce-paypal-payments
# Enhance the UX of the Pay Later and Vaulting features

**Issue**: #174

---

### Description

This PR implements the improvements suggested in #174:

- Disables the Vaulting checkbox only when vaulting isn't in the PayPal account's scope
- Displays the copy about enabling Vaulting in the PayPal account only when it's not in the account's scope
- Unchecks and disables the Pay Later checkboxes when Vaulting is enabled in the plugin, and vice versa
- Removes the admin notice mentioning that Pay Later is disabled due to Vaulting being enabled
- Removes the admin notice mentioning that Vaulting is disabled due to Pay Later being enabled
- Updates the UI dynamically following the selected options without having to reload the page

### Steps to test:

PayPal account's vaulting scope changes:
1. Disable vaulting on your PayPal account (I didn't find a way to do it on my sandbox account, so I made the method that checks whether vaulting is enabled in the scope return 'false' in the plugin)
2. Go to the WooCommerce -> Settings -> Payments -> PayPal -> Manage (or Set up if it isn't enabled)
3. Make sure the "Vaulting" checkbox is disabled, and that its label includes "To use vaulting features, you must enable vaulting on your account" 
4. Go to WooCommerce -> Settings -> Payments -> PayPal Card Processing -> Manage (or Set up if it isn't enabled)
5. Make sure step 3 is also true on this screen

1. Enable vaulting on your PayPal account
2. Go to the WooCommerce -> Settings -> Payments -> PayPal -> Manage (or Set up if it isn't enabled)
3. Make sure the "Vaulting" checkbox is enabled, and that its label is "Enable saved cards and subscription features on your store." alone, without "To use vaulting features, you must enable vaulting on your account" 
4. Go to WooCommerce -> Settings -> Payments -> PayPal Card Processing -> Manage (or Set up if it isn't enabled)
5. Make sure step 3 is also true on this screen

PayPal's settings page UI changes (do all of this without saving the settings):
1. Make sure Vaulting is enabled on your PayPal account
2. Go to the WooCommerce -> Settings -> Payments -> PayPal -> Manage (or Set up if it isn't enabled)
3. Make sure the Vaulting setting is disabled
4. Confirm it now has a description saying "This will disable all Pay Later messaging on your site"
5. Enable the "Enable message on Checkout", "Enable message on Single Product", and "Enable message on Cart" checkboxes
6. Confirm that the label for these checkboxes is "Enable on {location}"
7. Disable the Vaulting feature
8. Confirm that the checkboxes for "Enable message on Checkout", "Enable message on Single Product", and "Enable message on Cart" are now unchecked and disabled
9. Confirm that the labels for these settings changed from "Enable on {location}" to "You have PayPal vaulting enabled, that's why Pay Later Messaging options are unavailable now. You cannot use both features at the same time."
10. Confirm that their child settings are now hidden

This behavior should persist on each step when saving and reloading the page.

1. Make sure vaulting is enabled on your account
2. Go to the WooCommerce -> Settings -> Payments -> PayPal -> Manage (or Set up if it isn't enabled)
3. Disable vaulting on the plugin settings
4. Enable the Pay Later settings
5. Save the settings
6. Disable vaulting on your PayPal account
7. Go to the plugin's settings (or reload the page you were on)
8. Make sure the behavior is consistent with when the vaulting setting is disabled in the plugin

### Documentation

[This section](https://woocommerce.com/document/woocommerce-paypal-payments/#section-6) from WooCommerce docs would behave differently than what's described there after this PR is merged.
Under `To make Vault work with WooCommerce Subscriptions, follow the steps below.`, steps 2 and 3 won't be needed. You can now enable Vaulting without having to disable Pay Later and saving first.

### Changelog entry
> Enhance the UI around the Pay Later and Vaulting features.

Closes #174
